### PR TITLE
[ADD] New module partner_multi_company

### PR DESCRIPTION
**Pull request by oihane**

### CODE REVIEW

File: partner_multi_company/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Partner permissions for discrete companies
+==========================================
+
+This modules allows to select in which of the companies you want to use each
+of the products.
+
+Installation
+============
+
+This module uses the post and uninstall hooks for updating default product
```
Comment (pedrobaeza) [inline partner_multi_company/README.rst]: s/product/partner

File: partner_multi_company/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+template security rule. This only means that updating the module will not
+restore the security rule this module changes. Only a complete removal and
+reinstallation will serve.
+
+Usage
+=====
+
+On the product form view, go to the "Information" tab, and put the companies
```
Comment (pedrobaeza) [inline partner_multi_company/README.rst]: Change instructions for referring partners, not products

File: partner_multi_company/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# © 2015 Oihane Crucelaegui
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Partner multi-company",
+    "summary": "Select individually the partner visibility on each company",
+    "version": "8.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["base"],
+    "author": "Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline partner_multi_company/__openerp__.py]: Put "Serv. Tecnol. Avanzados - Pedro M. Baeza" and you if you want.

File: partner_multi_company/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "contributors": [
+        "Oihane Crucelaegui <oihanecruce@avanzosc.es>",
+    ],
+    "category": "Custom Module",
```
Comment (pedrobaeza) [inline partner_multi_company/__openerp__.py]: Isn't any other category?

File: partner_multi_company/hooks.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        for group in groups:
+            if not group['company_id']:
+                continue
+            partners = partner_model.search(group['__domain'])
+            partners.write(
+                {'company_ids': [(6, 0, [group['company_id'][0]])]})
+
+
+def uninstall_hook(cr, registry):
+    """Restore product rule to base value."""
```
Comment (pedrobaeza) [inline partner_multi_company/hooks.py]: s/product/partner

**Comment by pedrobaeza** [comment]
Now it's working.

By my part, :+1:

File: partner_multi_company/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        self._amend_company_id(vals)
+        # We must suspend security during this creation because it fails due to
+        # all the mail stuff in between that confuses security rules
+        obj = self.suspend_security()
+        return super(ResPartner, obj).create(vals)
+
+    @api.multi
+    @api.depends('company_ids')
+    def _compute_company_id(self):
```
Comment (JayVora-SerpentCS) [inline partner_multi_company/models/res_partner.py:55]: @api.one and @api.depends would be best, isn't it?

Comment (pedrobaeza) [inline partner_multi_company/models/res_partner.py:55]: `@api.one` is deprecated in v9, so I'm avoiding to use it directly. It's only one loop less...

**Comment by antoniocanovas** [comment]
Any problem with this module?
Functionalty 👍

**Comment by pedrobaeza** [comment]
Merging this old one that has 2 approvals.

**Pull request merged**